### PR TITLE
Booting grammar fixes

### PR DESCRIPTION
**Pull request by georgehorrell**
Some minor changes to the grammar in linux-bootstrap-1, hopefully improves the readability a little.

Cheers,
George

### CODE REVIEW

**Comment by Huddayn** [comment]
In my opinion , it doesn't imporve readability. And it also adds semantic errors in some places.

Regards

File: Booting/linux-bootstrap-1.md [hunk 1]
```diff
@@ -446,25 +450,25 @@ Ok now we have correct segment registers, stack, need only setup bss and jump to
...
-Jump to main
+Jump to Main
 --------------------------------------------------------------------------------
 
-That's all, we have stack, bss and now we can jump to `main` C function:
+That's all, we have the stack, bss and now we can jump to the `main` C function:
 
 ```assembly
-	calll main
+	calll  main
```
Comment (0xAX) [inline Booting/linux-bootstrap-1.md]: drop space here please

File: Booting/linux-bootstrap-1.md [hunk 2]
```diff
@@ -446,25 +450,25 @@ Ok now we have correct segment registers, stack, need only setup bss and jump to
 	rep; stosl
 ```
 
-First of all we put [__bss_start](https://github.com/torvalds/linux/blob/master/arch/x86/boot/setup.ld#L47) address in `di` and `_end + 3` (+3 - align to 4 bytes) in `cx`. Clear `eax` register with `xor` instruction and calculate size of BSS section (put in `cx`). Divide `cx` by 4 and repeat `cx` times `stosl` instruction which stores value of `eax` (it is zero) and increase `di`by the size of `eax`. In this way, we write zeros from `__bss_start` to `_end`:
+First of all we put [__bss_start](https://github.com/torvalds/linux/blob/master/arch/x86/boot/setup.ld#L47) address in `di` and `_end + 3` (+3 - align to 4 bytes) in `cx`. Next, we clear `eax` register with `xor` instruction and calculate the size of the BSS section (put in `cx`). Divide `cx` by 4 and repeat `cx` times `stosl` instruction which stores value of `eax` (it is zero) and increase `di` by the size of `eax`. In this way, we write zeros from `__bss_start` to `_end`:
 
 ![bss](http://oi59.tinypic.com/29m2eyr.jpg)
 
-Jump to main
+Jump to Main
```
Comment (0xAX) [inline Booting/linux-bootstrap-1.md]: `main` must be in lowercase, here, because function name

File: Booting/linux-bootstrap-1.md [hunk 3]
```diff
@@ -165,24 +169,24 @@ General real mode memory map is:
...
 ```
 
-At the start of execution BIOS is not in RAM, it is located in ROM.
+At the start of execution the BIOS is not in RAM, it is located in ROM.
 
 Bootloader
 --------------------------------------------------------------------------------
 
-Now BIOS has transferred control to the operating system bootloader and it needs to load operating system into the memory. There are a couple of bootloaders which can boot linux, like: [Grub2](http://www.gnu.org/software/grub/), [syslinux](http://www.syslinux.org/wiki/index.php/The_Syslinux_Project) and etc... Linux kernel has [Boot protocol](https://github.com/torvalds/linux/blob/master/Documentation/x86/boot.txt) which describes how to load linux kernel.
+Now the BIOS has transferred control to the operating system bootloader, it needs to load the operating system into memory. There are a number of bootloaders which can boot linux, including: [Grub2](http://www.gnu.org/software/grub/) and [syslinux](http://www.syslinux.org/wiki/index.php/The_Syslinux_Project). There is a linux [Boot protocol](https://github.com/torvalds/linux/blob/master/Documentation/x86/boot.txt) which describes how to load the linux kernel.
```
Comment (0xAX) [inline Booting/linux-bootstrap-1.md]: number?

File: Booting/linux-bootstrap-1.md [hunk 4]
```diff
@@ -130,26 +134,26 @@ We will see:
...
 
-**NOTE**: as you can read above the CPU is in real mode. In real mode, calculating the physical address in memory is as follows:
+**NOTE**: as you can see above, the CPU is in real mode. In real mode, calculating the physical address in memory is as follows:
 
 ```
 PhysicalAddress = Segment * 16 + Offset
 ```
 
-as I wrote above. But we have only 16 bit general purpose registers. The maximum value of 16 bit register is: `0xffff`; So if we take the biggest values, it will be:
+as we discussed above. But we have only 16 bit general purpose registers. The maximum value of 16 bit register is: `0xffff`; So if we take the biggest values, it will be:
```
Comment (0xAX) [inline Booting/linux-bootstrap-1.md]: hmm, i'm not sure, but i think that `wrote` was better, what do you think?

File: Booting/linux-bootstrap-1.md [hunk 5]
```diff
@@ -130,26 +134,26 @@ We will see:
...
+In this example we can see that this code will be executed in 16 bit real mode and will start at 0x7c00 in memory. After starting it calls the [0x10](http://www.ctyme.com/intr/rb-0106.htm) interrupt which just prints the `!` symbol. It fills the rest of the 510 bytes with zeros and finishes with two magic bytes 0xaa and 0x55.
 
-Real world boot loader starts at the same point, ends with `0xaa55` bytes, but reads kernel code from device, loads it to memory, parses and passes boot parameters to kernel and etc... instead of printing one symbol :) Ok, so, from this moment BIOS handed control to the operating system bootloader and we can go ahead.
+Real world boot loader starts at the same point, ends with `0xaa55` bytes, but reads kernel code from the device, loads it to memory, parses and passes boot parameters to kernel and etc... instead of printing one symbol :) Ok, so, from this moment on, BIOS has handed control to the operating system bootloader and we can proceed.
```
Comment (0xAX) [inline Booting/linux-bootstrap-1.md]: why is 'on' here?

I mean: `so, from this moment on`

File: Booting/linux-bootstrap-1.md [hunk 6]
```diff
@@ -1,29 +1,31 @@
...
-* Understanding C code
-* Understanding assembly code (AT&T syntax)
+* An understanding of C code
+* An understanding of assembly code (AT&T syntax)
 
-Anyway, if you just started to learn some tools, I will try to explain some parts during this and following posts. Ok, little introduction finished and now we can start to dive into kernel and low-level stuff.
+If you just started to learn some tools, I will try to explain some parts during this and the following posts. Ok, our little introduction is finished and now we can start to dive into th kernel and low-level stuff.
 
-All code is actual for kernel - 3.18, if there are changes, I will update posts.
+All code is actually for kernel - 3.18, if there are changes, I will update my posts.
```
Comment (0xAX) [inline Booting/linux-bootstrap-1.md]: why 'actually' here? I think that 'actual' is more suitable, or i'm wrong?

File: Booting/linux-bootstrap-1.md [hunk 7]
```diff
@@ -1,29 +1,31 @@
-Kernel booting process. Part 1.
+Kernel Booting Process. Part 1.
 ================================================================================
 
-From the bootloader to kernel
+From the Bootloader to the Kernel
```
Comment (0xAX) [inline Booting/linux-bootstrap-1.md:4]: What's the purpose to make it uppercase?

**Comment by 0xAX** [comment]
@georgehorrell first of all thank you for your work. I have left some comments.

@Huddayn thank you for review of this PR too... I left some comments, maybe i have missed something else?

File: Booting/linux-bootstrap-1.md [hunk 8]
```diff
@@ -302,9 +306,9 @@ _start:
 	//
 ```
 
-Here we can see `jmp` instruction opcode - `0xeb` to the `start_of_setup-1f` point. `Nf` notation means following: `2f` refers to the next local `2:` label. In our case it is label `1` which goes right after jump. It contains rest of setup [header](https://github.com/torvalds/linux/blob/master/Documentation/x86/boot.txt#L156) and right after setup header we can see `.entrytext` section which starts at `start_of_setup` label.
+Here we can see `jmp` instruction opcode - `0xeb` to the `start_of_setup-1f` point. `Nf` notation means following: `2f` refers to the next local `2:` label. In our case it is the label `1` which goes right after jump. It contains the rest of the setup [header](https://github.com/torvalds/linux/blob/master/Documentation/x86/boot.txt#L156) and right after the setup header we can see the `.entrytext` section which starts at `start_of_setup` label.
```
Comment (0xAX) [inline Booting/linux-bootstrap-1.md:309]: @Huddayn for me too, although i'm not english-man, all these a/the looks like magic for me:)

**Comment by Huddayn** [comment]
Ok, I've commented on the parts that I think should be changed.

**Comment by georgehorrell** [comment]
Huddayn, those are all very reasonable points; I'm not hugely familiar with technical writing but do have some experience in copywriting. Sorry about the numerous typos. 

How do I submit my correction to all of the errors?

File: Booting/linux-bootstrap-1.md [hunk 9]
```diff
@@ -1,29 +1,31 @@
...
 
 **Required knowledge**
 
-* Understanding C code
-* Understanding assembly code (AT&T syntax)
+* An understanding of C code
+* An understanding of assembly code (AT&T syntax)
 
-Anyway, if you just started to learn some tools, I will try to explain some parts during this and following posts. Ok, little introduction finished and now we can start to dive into kernel and low-level stuff.
+If you just started to learn some tools, I will try to explain some parts during this and the following posts. Ok, our little introduction is finished and now we can start to dive into th kernel and low-level stuff.
```
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md]: > Anyway, if you just started

The colloquialisms broke my flow when I was reading it over - I'd hate to take away from 0xAX's style though. Just a judgement call. 

> this and the following

This is correct. 'Following' means _in response to_ or _as a result of_ whereas 'the following' is used to introduce a noun or collective noun.

> the kernel

will correct this

File: Booting/linux-bootstrap-1.md [hunk 10]
```diff
@@ -423,19 +427,19 @@ If `CAN_USE_HEAP` bit is set, put `heap_end_ptr` to `dx` which points to `_end`
...
 ```assembly
 cmpl	$0x5a5aaa55, setup_sig
 jne	setup_bad
 ```
 
-just consists of comparing of [setup_sig](https://github.com/torvalds/linux/blob/master/arch/x86/boot/setup.ld#L39) and `0x5a5aaa55` number, and if they are not equal jump to error printing.
+just consists of comparing of [setup_sig](https://github.com/torvalds/linux/blob/master/arch/x86/boot/setup.ld#L39) and `0x5a5aaa55` number, and if they are not equal, jump to error printing.
 
-Ok now we have correct segment registers, stack, need only setup bss and jump to C code. Bss section used for storing statically allocated uninitialized data. Here is the code:
+Now we have correct segment registers, stack, we only to setup bss and jump to C code. Bss section used for storing statically allocated uninitialized data. Here is the code:
```
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md]: will correct

File: Booting/linux-bootstrap-1.md [hunk 11]
```diff
@@ -235,14 +239,14 @@ where `X` is the address kernel bootsector loaded. In my case `X` is `0x10000` (
...
 
-Start of kernel setup
+Start of Kernel Setup
 --------------------------------------------------------------------------------
 
-Finally we are in the kernel. Technically kernel didn't run yet, first of all we need to setup kernel, memory manager, process manager and etc... Kernel setup execution starts from [arch/x86/boot/header.S](https://github.com/torvalds/linux/blob/master/arch/x86/boot/header.S) at the [_start](https://github.com/torvalds/linux/blob/master/arch/x86/boot/header.S#L293). It is little strange at the first look, there are many instructions before it. Actually....
+Finally we are in the kernel. Technically, the kernel didn't run yet, first of all we need to setup the kernel, memory manager, process manager and etc... Kernel setup execution starts from [arch/x86/boot/header.S](https://github.com/torvalds/linux/blob/master/arch/x86/boot/header.S) at the [_start](https://github.com/torvalds/linux/blob/master/arch/x86/boot/header.S#L293). It is a little strange at first sight, as there are many instructions before it. But if we look back...
```
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md]: > the kernel didn't run yet

also I think this should read

> the kernel hasn't run yet

File: Booting/linux-bootstrap-1.md [hunk 12]
```diff
@@ -165,24 +169,24 @@ General real mode memory map is:
...
+Now the BIOS has transferred control to the operating system bootloader, it needs to load the operating system into memory. There are a number of bootloaders which can boot linux, including: [Grub2](http://www.gnu.org/software/grub/) and [syslinux](http://www.syslinux.org/wiki/index.php/The_Syslinux_Project). There is a linux [Boot protocol](https://github.com/torvalds/linux/blob/master/Documentation/x86/boot.txt) which describes how to load the linux kernel.
 
-Let us briefly consider how grub loads linux. GRUB2 execution starts from `grub-core/boot/i386/pc/boot.S`. It starts to load from device its own kernel (not to be confused with linux kernel) and executes `grub_main` after successfully loading.
+Let us briefly consider how grub loads linux. GRUB2 execution starts from `grub-core/boot/i386/pc/boot.S`. It starts by loading from the device, its own kernel (not to be confused with linux kernel) and then executes `grub_main`.
```
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md:185]: will correct

File: Booting/linux-bootstrap-1.md [hunk 13]
```diff
@@ -165,24 +169,24 @@ General real mode memory map is:
...
 
-But stop, at the beginning of post I wrote that first instruction executed by the CPU is located at address `0xfffffff0`, which is much bigger than `0xffff` (1MB). How can CPU access it in real mode? As I write about and you can read in [coreboot](http://www.coreboot.org/Developer_Manual/Memory_map) documentation:
+But wait a minute, at the beginning of the post I wrote that the first instruction executed by the CPU is located at address `0xfffffff0`, which is much bigger than `0xffff` (1MB). How can the CPU access it in real mode? As I write about and you can read in [coreboot](http://www.coreboot.org/Developer_Manual/Memory_map) documentation:
 
 ```
 0xFFFE_0000 - 0xFFFF_FFFF: 128 kilobyte ROM mapped into address space
 ```
 
-At the start of execution BIOS is not in RAM, it is located in ROM.
+At the start of execution the BIOS is not in RAM, it is located in ROM.
```
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md:178]: will correct

File: Booting/linux-bootstrap-1.md [hunk 14]
```diff
@@ -165,24 +169,24 @@ General real mode memory map is:
 0x000F0000 - 0x000FFFFF - System BIOS
 ```
 
-But stop, at the beginning of post I wrote that first instruction executed by the CPU is located at address `0xfffffff0`, which is much bigger than `0xffff` (1MB). How can CPU access it in real mode? As I write about and you can read in [coreboot](http://www.coreboot.org/Developer_Manual/Memory_map) documentation:
+But wait a minute, at the beginning of the post I wrote that the first instruction executed by the CPU is located at address `0xfffffff0`, which is much bigger than `0xffff` (1MB). How can the CPU access it in real mode? As I write about and you can read in [coreboot](http://www.coreboot.org/Developer_Manual/Memory_map) documentation:
```
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md:172]: will correct

File: Booting/linux-bootstrap-1.md (hunk 6 above)
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md]: does `actual` have a technical meaning? This might be better as

> All code is written for kernel - 3.18

File: Booting/linux-bootstrap-1.md (hunk 3 above)
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md]: > There are couple of bootloaders which can boot linux, including: Grub2 and syslinux.

missing an 'a' before couple? I avoided couple because there are [more than two](http://en.wikipedia.org/wiki/Comparison_of_boot_loaders) bootloaders.

> Linux kernel has Boot protocol which describes how to load the linux kernel.

This doesn't make sense to me. In the documentation it is referred to as _the linux/x86 boot protocol_, your rephrasing would suggest that the kernel is loading itself.

File: Booting/linux-bootstrap-1.md (hunk 1 above)
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md]: will correct

File: Booting/linux-bootstrap-1.md (hunk 2 above)
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md]: will correct

File: Booting/linux-bootstrap-1.md (hunk 4 above)
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md]: stylistic preference. I will change it back.

File: Booting/linux-bootstrap-1.md (hunk 7 above)
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md:4]: Capitalising the nouns is normally how you format a title, if you would rather not I will change those back also.

File: Booting/linux-bootstrap-1.md [hunk 15]
```diff
@@ -97,7 +101,7 @@ SECTIONS {
 }
 ```
 
-Now BIOS has started to work. After all initializations and hardware checking, it needs to load operating system. BIOS tries to find bootable device which contains boot sector. Boot sector is the first sector on device (512 bytes) and contains sequence of `0x55` and `0xaa` at 511 and 512 byte. For example:
+Now BIOS has started to work. After all initializations and hardware checking, it needs to load the operating system. BIOS tries to find a bootable device which contains a boot sector. A boot sector is the first sector on the device (512 bytes) and contains a sequence of `0x55` and `0xaa` at 511 and 512 byte. For example:
```
Comment (georgehorrell) [inline Booting/linux-bootstrap-1.md:104]: I'm slowly getting more familiar with where to insert the articles in this type of documentation - thanks for correcting this for me.

**Comment by Huddayn** [comment]
@georgehorrell Just make another commit to the same fork.

File: Booting/linux-bootstrap-1.md (hunk 15 above)
Comment (ccope) [inline Booting/linux-bootstrap-1.md:104]: BIOS isn't a proper noun, it's an acronym for a common noun (there are numerous BIOSes). Therefore, it should be referred to as 'the BIOS'.

File: Booting/linux-bootstrap-1.md [hunk 16]
```diff
@@ -130,26 +134,26 @@ We will see:
 
 ![Simple bootloader which prints only `!`](http://oi60.tinypic.com/2qbwup0.jpg)
 
-In this example we can see that this code will be executed in 16 bit real mode and will start at 0x7c00 in memory. After the start it calls the [0x10](http://www.ctyme.com/intr/rb-0106.htm) interrupt which just prints `!` symbol. It fills rest of 510 bytes with zeros and finish with two magic bytes 0xaa and 0x55.
+In this example we can see that this code will be executed in 16 bit real mode and will start at 0x7c00 in memory. After starting it calls the [0x10](http://www.ctyme.com/intr/rb-0106.htm) interrupt which just prints the `!` symbol. It fills the rest of the 510 bytes with zeros and finishes with two magic bytes 0xaa and 0x55.
```
Comment (ccope) [inline Booting/linux-bootstrap-1.md:137]: Instead of '0x7c00 in memory' I would write 'memory address `0x7c00`'

File: Booting/linux-bootstrap-1.md (hunk 14 above)
Comment (ccope) [inline Booting/linux-bootstrap-1.md:172]: CPU is an acronym for another common noun (central processing unit). The use of 'the' is correct.

File: Booting/linux-bootstrap-1.md (hunk 13 above)
Comment (ccope) [inline Booting/linux-bootstrap-1.md:178]: This 'the' is correct.

File: Booting/linux-bootstrap-1.md (hunk 5 above)
Comment (ccope) [inline Booting/linux-bootstrap-1.md]: The wording there is a bit awkward, `on` and `proceed` shouldn't be used together like this. `At this moment, the BIOS has handed control to the bootloader.` Please revert the change to this line though, I've updated the bootloader section in another pull request.

File: Booting/linux-bootstrap-1.md (hunk 6 above)
Comment (ccope) [inline Booting/linux-bootstrap-1.md]: There are a couple of ways to phrase this. I believe you originally intended to say `All kernel code in this document is actual code from Linux 3.18`. I think `This document contains sample code from Linux 3.18.` is better.

File: Booting/linux-bootstrap-1.md (hunk 15 above)
Comment (ccope) [inline Booting/linux-bootstrap-1.md:104]: I actually added some detail to this section in my pull request.

File: Booting/linux-bootstrap-1.md (hunk 12 above)
Comment (ccope) [inline Booting/linux-bootstrap-1.md:185]: I rewrote this section in my PR.

**Comment by ccope** [comment]
@georgehorrell  if you're working in the web interface, go here and click the edit button: https://github.com/georgehorrell/linux-insides/blob/booting-grammar-fixes/Booting/linux-bootstrap-1.md
Then just commit your changes on the same branch, and they will show up here.

If you're on the command line, just make the changes, `git add` the filename, `git commit`, and `git push`.

**Comment by 0xAX** [comment]
@georgehorrell thank you that you updated PR.

I can't merge it without conflicts. Can you please rebase on must. If you will need any help with this, let me know

**Comment by 0xAX** [comment]
@caffinatedmonkey I know :) But It will be easier to review it. @georgehorrell please rebase on master and I will review and merge it.

Thank you
